cdap-ui/app/features/hydratorplusplus/services/hydrator-node-service.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+class HydratorPlusPlusNodeService {
+  constructor($q, HydratorPlusPlusHydratorService, IMPLICIT_SCHEMA) {
+    this.$q = $q;
+    this.HydratorPlusPlusHydratorService = HydratorPlusPlusHydratorService;
+    this.IMPLICIT_SCHEMA = IMPLICIT_SCHEMA;
+  }
+  getPluginInfo(node, appType, sourceConn) {
+    var promise;
+    if (node._backendProperties) {
+      promise = this.$q.when(node);
+    } else {
+      promise = this.HydratorPlusPlusHydratorService.fetchBackendProperties(node, appType);
+    }
+    return promise.then((node) => this.configurePluginInfo(node, appType, sourceConn));
+  }
+  configurePluginInfo(node, appType, sourceConn) {
+      var input;
+      var sourceSchema = null;
+      var isStreamSource = false;
+
+      var clfSchema = this.IMPLICIT_SCHEMA.clf;
+
+      var syslogSchema = this.IMPLICIT_SCHEMA.syslog;
+
+      var source;
+      if (sourceConn && sourceConn.length) {
+        source = sourceConn[0];
+        sourceSchema = source.outputSchema;
+
+        if (source.plugin.name === 'Stream') {
+          isStreamSource = true;
+        }
+
+        if (source.plugin.properties.format && source.plugin.properties.format === 'clf') {
+          sourceSchema = clfSchema;
+        } else if (source.plugin.properties.format && source.plugin.properties.format === 'syslog') {
+          sourceSchema = syslogSchema;
+        }
```

Comment:
instead of doing this explicitly, we can do an ```Object.keys(this.IMPLICIT_SCHEMA)``` to get the list of available implicit schema, then just simply set ```sourceSchema = this.IMPLICIT_SCHEMA[source.plugin.properties.format];```